### ISSUE TITLE
Prettier keeps breaking .babelrc

### ISSUE THREAD

**Issue by bradennapier** [prettier/prettier#4636]
Hard to do the playground - but prettier is completely breaking my system because it makes .babelrc invalid

Input (.babelrc):
```json
{
    "presets": [
      "@babel/preset-flow",
      [
        "@babel/preset-env",
        {
          "cacheDirectory": true,
          "useBuiltIns": "usage",
          "targets": {
            "node": "8.10"
          }
        }
      ],
      ["@babel/preset-stage-0", {
        "decoratorsLegacy": true
      }]
    ],
    "plugins": ["lodash", "@babel/plugin-codemod-optional-catch-binding"]
  }
  ```

Output
```
{
  presets: [
    '@babel/preset-flow',
    [
      '@babel/preset-env',
      {
        cacheDirectory: true,
        useBuiltIns: 'usage',
        targets: {
          node: '8.10',
        },
      },
    ],
    [
      '@babel/preset-stage-3',
      {
        decoratorsLegacy: true,
      },
    ],
  ],
  plugins: ['lodash', '@babel/plugin-codemod-optional-catch-binding'],
}
```

Obviously .babelrc is not a js file and it is not indicated as so yet it appears to assume it is.  This only happens specifically for `.babelrc` - not sure why it assuming its js?

In babel 7 they optionally will allow `.babelrc.js` but this does not have such a file extension.

**Comment by suchipi** [comment]
`.babelrc` is a [JSON5](https://github.com/json5/json5) file, so this is still valid. Does babel not work correctly after prettier makes this change?

**Comment by bradennapier** [comment]
The editor doesnt like it, that's for sure so it makes this type of thing with the most popular editor (and atom is the same).  

![image](https://user-images.githubusercontent.com/15365418/40944226-d802c7f8-6808-11e8-9f6e-3335f6a8cf2e.png)

Since they default to JSON not JSON5

**Comment by suchipi** [comment]
@ikatyang based on this issue and https://github.com/prettier/prettier/issues/4611 I think we should *not* convert Object keys to identifiers by default in the json5 printer; it is surprising behavior to some. Thoughts?

**Comment by bradennapier** [comment]
Would definitely be nice if the editor knew `.babelrc` is actually JSON5 by default - not sure if there is the possibility of detecting if JSON5 is used and basing it on that (look for comments/trailing commands/non-quoted keys) - but that seems like a good amount extra work.

**Comment by j-f1** [comment]
@bradennapier Is that VS Code or Atom?

**Comment by azz** [comment]
VS Code

**Comment by ikatyang** [comment]
It seems this issue is solved?

@suchipi I'm not sure what should we do here since converting object keys to identifiers is valid in JSON5, and we don't want to add options. And also if we're not going to convert them, there's no difference between JSON and JSON5, thus no need to split JSON5 from JSON.

And based on the response from linguist, we're probably going to add `.jsonc` to JSON's extensions for #4611.

(It's really weird to me why VSCode/Atom didn't have the built-in JSON5 support.)

**Comment by suchipi** [comment]
I'll move discussion re: quoted keys to a new issue

**Comment by buffos** [comment]
Actually when the I have the quoted version of my .babelrc and have an env section like the one below

`"env": {
    "test": {
      "plugins": ["transform-es2015-modules-commonjs"]
    }
  }`

and then run with `set BABEL_ENV="test"&&command.... ` it will not work (or without the quotes around test)

It works fine in the unquoted version.

**Comment by suchipi** [comment]
@buffos what command are you running? That sounds like a babel issue.

**Comment by buffos** [comment]
@suchipi 

just running jest from the command line

`set BABEL_ENV="test"&&.\node_modules\.bin\jest`

**Comment by lydell** [comment]
Assuming bash, I don't think you are supposed to use `set` there. As far as I remember, `set` is used to set "bash settings", not for setting environment variables. I think you're looking for `BABEL_ENV=test command...`.

**Comment by lydell** [comment]
Assuming bash, I don't think you are supposed to use `set` there. As far as I remember, `set` is used to set "bash settings", not for setting environment variables. I think you're looking for `BABEL_ENV=test command...`.

**Comment by buffos** [comment]
@lydell 
No its windows. Tried both with and without quotes. Will not work.
When I prettier changes to the non-quoted version of the rc file, everything works fine

**Comment by ikatyang** [comment]
@macrozone FYI, it should be fixed in #4734, which treats `.babelrc` as `JSON with Comments` (github/linguist#4171) and uses `--parser json` by default. As a temporary workaround you could specify the parser manually in the config file for now:

```
{
  "overrides": [
    {
      "files": ".babelrc",
      "options": {
        "parser": "json"
      }
    }
  ]
}
```
